microbench/src/main/java/io/netty/microbench/concurrent/BurstCostExecutorsBenchmark.java

```diff
@@ -0,0 +1,337 @@
+ * the License.
+ */
+package io.netty.microbench.concurrent;
+
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.kqueue.KQueue;
+import io.netty.channel.kqueue.KQueueEventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.concurrent.DefaultEventExecutor;
+import io.netty.util.internal.PlatformDependent;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class BurstCostExecutorsBenchmark extends AbstractMicrobenchmark {
+
+    /**
+     * This is executor is useful as baseline because it won't go to sleep and won't be hit by the cost of being awaken
+     * on both offer/consumer side.
+     */
+    private static final class SpinExecutorService implements ExecutorService {
+
+        private static final Runnable POISON_PILL = new Runnable() {
+            @Override
+            public void run() {
+            }
+        };
+        private final Queue<Runnable> tasks;
+        private final AtomicBoolean poisoned = new AtomicBoolean();
+        private final Thread executorThread;
+
+        public SpinExecutorService(int maxTasks) {
+            tasks = PlatformDependent.newFixedMpscQueue(maxTasks);
+            executorThread = new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    final Queue<Runnable> tasks = SpinExecutorService.this.tasks;
+                    while (true) {
+                        final Runnable task = tasks.poll();
+                        if (task != null) {
+                            if (task == POISON_PILL) {
+                                return;
+                            } else {
+                                task.run();
+                            }
+                        }
+                    }
```

Comment:
nit this could be simplified:

```java
Runnable task;
while ((task = tasks.poll()) != POISON_PILL) {
    if (task != null) {
        task.run();
    }
}
```